### PR TITLE
develop/app

### PR DESCRIPTION
**Pull request by Muskan-Mittal**
This is an app to set up alarms for medicines targetting the aged population.
- The user will sign up or log in to the app, which is done using firebase. Log in using Facebook and Goole sign-in is also added for good user experience.
- Add Medicine button allows adding a medicine asking the user about the details of consumption.
- My Medicines displays the entire list of medicines added by the user. Card view is used for setting the UI. The user can click on the card view to update the schedule for the medicine.
- Reminders display the list of the active reminders.[WIP]
- Viewpager with tabs is used for better user experience for the timeline and reminders.
- Logout functionality is also added for the user to get successfully logged out.
update_medicine
my_medicines
reminders_tab
add_medicine
login
dashboard
sign_up
first_screen

### CODE REVIEW

**Comment by MehaKaushik** [comment]
The comment space where you mentioned "review required" is meant for the reviewer to know what does this PR do. 
Format for sending PR should be:
1. What feature did you add ? 
2. Why did you add this feature ? (answers could be: good user experience, optimised algo etc etc)
3. Proof that the feature you added works as expected.( Some screenshot in case of static UI changes, gifs in case of functionalities)

Please note: One PR should contain one change only. If circumstances do not allow so, then, one commit should have one feature only.  In such cases, the above 3 points, should be written for each of the commit.

File: .idea/gradle.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (MehaKaushik) [inline .idea/gradle.xml:1]: These automatically generated files should be pushed directly as a part of "initial commit" . PRs are sent for the actual code written by a developer.

File: README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Medical_Helper
```
Comment (MehaKaushik) [inline README.md:1]: Please add a Readme. 
Resources : [1] https://gist.github.com/PurpleBooth/109311bb0361f32d87a2
[2] https://github.com/matiassingers/awesome-readme
 
Note : Please add this task of adding Readme, as a task in LIG website.

File: app/src/test/java/com/example/muskan/medical_help/ExampleUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * @see <a href="http://d.android.com/tools/testing">Testing documentation</a>
+ */
+public class ExampleUnitTest {
```
Comment (MehaKaushik) [inline app/src/test/java/com/example/muskan/medical_help/ExampleUnitTest.java:12]: Why is this unit test added ?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<resources>
+    <string name="app_name">Medical Helper</string>
+    <string name="email">email</string>
+    <string name="UserId">User Id</string>
+    <string name="Password">Password</string>
+    <string name="ConfirmPassword">Confirm Password</string>
+    <string name="error_message_email">Enter Valid Email</string>
```
Comment (MehaKaushik) [inline app/src/main/res/values/strings.xml]: Please be consistent in naming of variables. ConfirmPassword is camelcased and error_message_email is dash separated. Camel casing is widely popular, I recommend you to use it. 
refer: https://en.wikipedia.org/wiki/Camel_case

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    <string name="error_message_password">Enter Password</string>
+    <string name="error_password_match">Password doesnot match</string>
+    <string name="facebook_app_id">2030765153863126</string>
+    <string name="fb_login_protocol_scheme">fb2030765153863126</string>
+    <string name="action_emergency">Emergency</string>
+    <string name="action_settings">Settings</string>
+    <string name="action_aboutus">About Us</string>
+    <string name="action_logout">Logout</string>
+
```
Comment (MehaKaushik) [inline app/src/main/res/values/strings.xml]: Please do not leave blank lines. Blank lines are used to logically separate piece of codes, 
example : 
Wrong
int a;
int b; 
int c;
a = b+c;
c = a-b;
char f;


char g;


Right : 
int a;
int b;
int c;

a = b+c;
c = a-b;

char f;
char g;

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    <string name="navigation_drawer_close">Close navigation drawer</string>
+    <string name="openDrawer">open_drawer</string>
+    <string name="closeDrawer">close_drawer</string>
+
+    <!-- Navigation Drawer menu labels -->
+    <string name="nav_emergency">Emergency</string>
+    <string name="nav_settings">Settings</string>
+    <string name="nav_aboutus">About Us</string>
+    <string name="nav_logout">Logout</string>
+    <!-- Toolbar titles when navigation menu item is selected -->
```
Comment (MehaKaushik) [inline app/src/main/res/values/strings.xml]: Add a blank line after line 28.

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,45 @@
+<resources>
+    <string name="app_name">Medical Helper</string>
```
Comment (MehaKaushik) [inline app/src/main/res/values/strings.xml]: Please add a comment explaining where these variables are used.

File: app/src/main/res/layout/activity_signup.xml [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+            android:id="@+id/login_tv"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text=" LOGIN"
+            android:textColor="#ef5350"/>
+
+    </LinearLayout>
+
+</LinearLayout>
+
```
Comment (MehaKaushik) [inline app/src/main/res/layout/activity_signup.xml]: Please remove these blank lines.

File: app/src/main/res/layout/activity_signup.xml [hunk 9]
```diff
@@ -0,0 +1,188 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (MehaKaushik) [inline app/src/main/res/layout/activity_signup.xml:1]: Please check the formatting of this complete file.

File: app/src/main/res/layout/activity_signup.xml [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+
+    <LinearLayout
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:layout_marginBottom="15dp"
+        android:layout_marginTop="18dp"
+        android:orientation="horizontal">
+
+        <!--Facebook button-->
```
Comment (MehaKaushik) [inline app/src/main/res/layout/activity_signup.xml]: Add comments like this, in the complete file. Please do not add unnecessary comments.

File: app/src/main/res/layout/activity_main.xml [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    android:theme="@style/AppTheme">
+
+    <!--App logo-->
+    <ImageView
+        android:layout_width="150dp"
+        android:layout_height="150dp"
+        android:layout_gravity="center"
+        android:layout_marginTop="100dp"
+        android:src="@drawable/logo"/>
+
```
Comment (MehaKaushik) [inline app/src/main/res/layout/activity_main.xml:18]: Add a small comment here.

File: app/src/main/res/layout/activity_main.xml [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="30dp"
+        android:text="Having trouble in remembering your medicines? Set alarms and also save your medical reports online through DR. Help"
+
+        android:textColor="#212121"
+        android:textSize="16sp"/>
+
```
Comment (MehaKaushik) [inline app/src/main/res/layout/activity_main.xml]: Please add a small comment here. Please check the same throughout in this whole PR.

File: app/src/main/java/com/example/muskan/medical_help/user_model.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
+package com.example.muskan.medical_help;
+
+public class user_model {
+
+    private int id;
+    private String userId;
+    private String pwd;
```
Comment (MehaKaushik) [inline app/src/main/java/com/example/muskan/medical_help/user_model.java]: "pwd" is not a descriptive name, please use the complete word "password"

**Review by MehaKaushik — COMMENTED** [review]
Would send another set of comments for java files. Please have a look at these set of comments.

File: .idea/gradle.xml (hunk 1 above)
Comment (Muskan-Mittal) [reply .idea/gradle.xml:1]: Can anything be done now?

**Review by MehaKaushik — APPROVED** [review]

**Pull request merged**